src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/Runtime/BindingFlagSupport/QueryResult.cs

```diff
@@ -112,11 +112,11 @@ public void CopyTo(MemberInfo[] array, int startIndex)
                         // declared by the most derived type. Since QueriedMemberLists are sorted in order of decreasing derivation,
                         // that means we let the first match win - unless, of course, they're both the "most derived member".
                         if (match.DeclaringType!.Equals(challenger.DeclaringType))
-                            throw new AmbiguousMatchException();
+                            throw ThrowHelper.GetAmbiguousMatchException(match);
```

Comment:
Could merge these lines like we do in `QueryResult.cs`. Thoughts as to which pattern looks better?